components/scream/src/physics/p3/tests/p3_unit_tests.cpp

```diff
@@ -80,6 +79,143 @@ struct UnitWrap::UnitTest<D>::TestP3Func
   }
 };
 
+template <typename D>
+struct UnitWrap::UnitTest<D>::TestP3Conservation
+{
+
+  KOKKOS_FUNCTION static void cloud_water_conservation_tests(int& errors){
+
+    Spack qc(1e-5);
+    Spack qcnuc(0.0);
+    Scalar dt = 1.1;
+    Spack qcaut(1e-4);
+    Spack qcacc(0.0);
```

Comment:
I see a lot of zeros in the inputs; is there any way we can get more interesting input data?

Reply:
We could add more interesting input data, but it wouldn't really add anything to the test other than the complexity of coming up with physical values. These conservation corrections aren't really physical, so it didn't seem like it was necessary to use more physical values.  But I'm open to improving this.

Reply:
OK, if you think it's fine then I'm fine with it too.

Reply:
I think the way these tests work is that if the sum of a bunch of depletion processes times the timestep is greater than the amount of mass you started with, those depletion processes are rescaled so you end up with zero (but not negative) mass at the end of the timestep. Because those depletion processes are only used in a sum, Kyle's approach of randomly choosing one to be big and all the others to be zero should work fine.